src/schoology-extractor/schoology_extractor/mapping/submissions.py

```diff
@@ -0,0 +1,80 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+from datetime import datetime
+
+import pandas as pd
+
+from . import constants
+
+
+def map_to_udm(submissions_df: pd.DataFrame) -> pd.DataFrame:
+    """
+    Maps a DataFrame containing Schoology submissions into the Ed-Fi LMS Unified Data
+    Model (UDM) format.
+
+    Parameters
+    ----------
+    assignments_df: DataFrame
+        Pandas DataFrame containing Schoology submissions for a section
+
+    Returns
+    -------
+    DataFrame
+        A LMSUsers-formatted DataFrame
+
+    Notes
+    -----
+    DataFrame columns are:
+        SourceSystemIdentifier: A unique number or alphanumeric code assigned to a submission by the source system
+        SourceSystem: The system code or name providing the user data
+        SubmissionStatus: The status of the submission
+        SubmissionDateTime: Datetime of the submission
+        EarnedPoints: Earned points for the submission
+        Grade: Grade for the submission
+        AssignmentSourceSystemIdentifier: Unique identifier for the assignment
+        LMSUserSourceSystemIdentifier: Unique identifier of the LMSUser
+        EntityStatus: Status of the entity
+        CreateDate: Created date
+        LastModifiedDate: Last modified date
+    """
+
+    if submissions_df.empty:
+        return submissions_df
+
+    df = submissions_df[
+        [
+            "id",
+            "created",
+            "late",
+            "draft",
+            "uid",
+            "CreateDate",
+            "LastModifiedDate"
+        ]
+    ].copy()
+
+    df["SourceSystem"] = constants.SOURCE_SYSTEM
+    df["EntityStatus"] = constants.ACTIVE
+    df["SubmissionStatus"] = df["late"].apply(lambda x: 'late' if x == 1 else '')
+    df["SubmissionStatus"] = df["draft"].apply(lambda x: 'draft' if x == 1 else x)
```

Comment:
Is the `SubmissionStatus` going to be blank if it is not late or draft? Seems like it would be better to write "complete" or "on-time" here instead of leaving an ambiguous blank.

Reply:
✅